docs/fields.md

```diff
@@ -5122,6 +5123,24 @@ ObjectReference contains enough information to let you inspect or modify the ref
 |`resourceVersion`|`string`|Specific resourceVersion to which this reference is made, if any. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency|
 |`uid`|`string`|UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids|
 
+## Duration
+
+Duration is a wrapper around time.Duration which supports correctmarshaling to YAML and JSON. In particular, it marshals into strings, whichcan be used as map keys in json.
+
+<details>
+<summary>Examples with this field (click to open)</summary>
+<br>
+
+- [`pod-gc-strategy-with-label-selector.yaml`](https://github.com/argoproj/argo-workflows/blob/master/examples/pod-gc-strategy-with-label-selector.yaml)
+
+- [`pod-gc-strategy.yaml`](https://github.com/argoproj/argo-workflows/blob/master/examples/pod-gc-strategy.yaml)
+</details>
+
+### Fields
+| Field Name | Field Type | Description   |
+|:----------:|:----------:|---------------|
+|`duration`|`string`|_No description available_|
+
```

Comment:
this is the part I find issues with from the codegen. It's missing spaces (between multi-line comments in the Go), for one thing, but also it confusingly lists `duration` as a sub-field. That is correct for the Go type itself, but it marshals to a string, so the configuration is just a plain string.

This doc makes me think I would do:

```yaml
  podGC:
    deleteDelayDuration:
      duration: 30s
```

instead of just `deleteDelayDuration: 30s`, which is how it actually is.

Is there a way to override this or something?

Reply:
Well, I was able to fix the spacing by modifying [this line](https://github.com/argoproj/argo-workflows/blob/82310dd459aa080754169d6a1667d30d9b7c75bf/hack/docgen.go#L60) in `docgen.go`, but while fixing this it ends up adding extraneous spaces in several places 😕
The resulting diff for `fields.md` ends up quite large as well, so I didn't even commit that change

Reply:
> This doc makes me think I would do:

This also affects the OpenAPI schema, not just `fields.md`, per #12631. The CRDs were correct though per https://github.com/argoproj/argo-workflows/issues/12631#issuecomment-1931128000 as the k8s tooling knows [how to handle this specific type](https://github.com/kubernetes/apimachinery/blob/2465dc5239ab8827a637148a78b380c278b4a5f4/pkg/apis/meta/v1/duration.go#L65).

I wrote up a workaround for this in #13543 

> Is there a way to override this or something? Casting somewhere? Not sure

I tried casting to a string in the `proto` struct tags of the type with `casttype`, but that didn't work and caused errors. It's possible I was doing it wrong though 🤷